code/modules/byvue/ui.dm

```diff
@@ -0,0 +1,119 @@
+/*
+Byond Vue UI framework
+main ui datum.
+*/
+/datum/byvueui
+    // title of ui window
+    var/title = "HTML5 UI"
+    // user that opened this ui
+    var/mob/user
+    // object that contains this ui
+    var/datum/object
+    // apperant object that contains ui, used for checks
+    var/atom/wobject = null
```

Comment:
What's the difference between `object` and `wobject`?

Reply:
object is the one receiving calls, and managing data, wobject is optional supplemented object that is present in world and it's coordinates can be checked like, if it is near user. if object is already an atom, then it should be used for checks, if it is not, then wobject. if both are unavailable for checks, then it should be used as 'admin' ui.